test/clustertest/tests/ClusterUpgradeTest.cpp

```diff
@@ -0,0 +1,172 @@
+#include <sys/stat.h>
+#include <test/clustertest/BedrockClusterTester.h>
+
+struct ClusterUpgradeTest : tpunit::TestFixture {
+    ClusterUpgradeTest()
+        : tpunit::TestFixture("ClusterUpgrade",
+                              BEFORE_CLASS(ClusterUpgradeTest::setup),
+                              AFTER_CLASS(ClusterUpgradeTest::teardown),
+                              TEST(ClusterUpgradeTest::test)
+                             ) { }
+
+    BedrockClusterTester* tester;
+    string prodBedrockName;
+
+    void setup() {
+        // Get the most recent releases.
+        const size_t RECENT_RELEASES_TO_CHECK = 5;
+        const string tempJson = "brdata.json";
+        string command = "curl --silent 'https://api.github.com/repos/Expensify/Bedrock/releases?page=1&per_page=" + to_string(RECENT_RELEASES_TO_CHECK) + "' -o " + tempJson;
+        ASSERT_EQUAL(system(command.c_str()), 0);
+
+        // Parse the JSON we got from github with recent releases.
+        string data = SFileLoad(tempJson);
+        SFileDelete(tempJson);
+        string bedrockTagName;
+        list<string> j1 = SParseJSONArray(STrim(data));
+        ASSERT_EQUAL(j1.size(), RECENT_RELEASES_TO_CHECK);
+
+        // We need the tag names from each release.
+        array<string, RECENT_RELEASES_TO_CHECK> tagNames;
+        for (size_t i = 0; i < RECENT_RELEASES_TO_CHECK; i++) {
+            STable j2 = SParseJSONObject(j1.front());
+            auto tag = j2.find("tag_name");
+            if (tag != j2.end()) {
+                tagNames[i] = tag->second;
+            }
+            if (j1.size()) {
+                j1.pop_front();
+            }
+        }
+
+        // Now choose the one to use. We want to test against the msot recent release that isn't the commit we're currently on.
+        // the commit number of the tag: git rev-list -n 1 $TAG
+        // The commit number we're currently on: git rev-parse HEAD
+        // If the current commit matches the tested tag, the script returns 1 and we check the next one. When the script returns 0, that's the release we'll use.
+        for (const auto& tagName : tagNames) {
+            string checkIfOnLatestTag = "/bin/bash -c 'if [ \"`git rev-list -n 1 " + tagName + "`\" = \"`git rev-parse HEAD`\" ]; then exit 1; else exit 0; fi'";
+            int result = system(checkIfOnLatestTag.c_str());
+            if (!result) {
```

Comment:
NAB, I find this easier to read

```suggestion
            if (result == 0) {
```